### ISSUE TITLE
Timezone handling error in DateUtils.mixedDateToDate()

### ISSUE THREAD

**Issue by dario1985** [typeorm/typeorm#1694]
**Issue type:**

[ ] question
[x ] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[x ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `websql`

**TypeORM version:**

[x ] `latest`
[ ] `@next`
[x ] `^0.1.13` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

```
const datestr = '2018-03-01T10:00:00.000Z' // with Timezone explicit to UTC!
DateUtils.mixedDateToDate(datestr)
> 2018-03-01T10:00:00.000Z
DateUtils.mixedDateToDate(datestr, true) // e.g. used on MySQL
> 2018-03-01T09:00:00.000Z
DateUtils.mixedDateToDate(datestr) === DateUtils.mixedDateToDate(datestr, true)
> false // should be true
```

What is the use of the [`toUtc`](https://github.com/typeorm/typeorm/blob/82b1410f0e4d7e481f596874fa5c303e3c50019c/src/util/DateUtils.ts#L47) flag, when should apply? It can cause double timezone conversions issues

**Comment by dario1985** [comment]
Nice! I think this it's pretty critical as it affects all no-utc systems. 
When is expected next branch to be released? May this fix can be released in the current 1.x too!

**Comment by AlexMesser** [comment]
Dates behavior changed in `0.2.0-alpha.29`. Please check it via `npm i typeorm@next`

**Comment by dario1985** [comment]
IHMO this is not a feature... it's a bug and it should be fixed also in the current stable branch too (as people may want to use it on production instead of a next/alpha lol)

**Comment by pleerock** [comment]
Released in `0.1.18`. Let us know if you still have issues.
